planner/planner_APIs.go

```diff
@@ -513,6 +512,11 @@ func (planner *MyPlanner) getPlanDetails(c *gin.Context) {
 	}
 	iowrappers.Logger.Debugf("Trip Details:\n%v\n", tripResp.PlaceDetails)
 
+	jsonOnly := strings.ToLower(c.DefaultQuery("json_only", "false"))
+	if jsonOnly != "false" {
+		c.JSON(http.StatusOK, tripResp)
+		return
+	}
```

Comment:
Can we convert the string to boolean in JS here?

Reply:
The `json_only` is from query parameter list of query URL "url?json_only=true". So the datatype may be string by default once we parse it in Go. Do you suggest us converting it to boolean in Go here?

Reply:
I am not sure if we could convert a string to boolean in Go?

Reply:
I suppose `jsonOnly != "false"` somewhat does this datatype conversion in Go.

Reply:
Similar usage of `jsonOnly` can be seen in line 465.